src/diart/models.py

```diff
@@ -38,11 +51,17 @@ def load(self):
 
     def to(self, *args, **kwargs) -> nn.Module:
         self.load()
-        return super().to(*args, **kwargs)
+        return self.model.to(*args, **kwargs)
 
     def __call__(self, *args, **kwargs):
         self.load()
-        return super().__call__(*args, **kwargs)
+        return self.model(*args, **kwargs)
+
+    def eval(self) -> "LazyModel":
+        self.load()
+        if not isinstance(self.model, WeSpeakerPretrainedSpeakerEmbedding):
```

Comment:
This check is way too specific, it will not work for other types of embeddings. I would check with `if isinstance(self.model, nn.Module)`

Reply:
I really like the idea, will add it!